Telemetry changes (#186)
meilisearch/specifications: text/0034-telemetry-policies.md

| Documents Searched GET | Aggregated event on all received requests via the `GET` - `/indexes/:indexUid/search` route during one hour or until a batch size reaches `500Kb`. |
| Documents Added | Aggregated event on all received requests via the `POST` - `/indexes/:indexUid/documents` route during one hour or until a batch size reaches `500Kb`. |
| Documents Updated | Aggregated event on all received requests via the `PUT` - `/indexes/:indexUid/documents` route during one hour or until a batch size reaches `500Kb`. |

> | Documents Deleted | Aggregated event on all received requests via the `DELETE` - `/indexes/:indexUid/documents`, `DELETE` - `/indexes/:indexUid/documents/:documentId`, `POST` - `indexes/:indexUid/documents/delete-batch` routes during one hour or until a batch size reaches `500Kb`.  |

Reviewer: Super meta comment.
Do you think we should start to send analytics on our analytics?
Like, send an event when a batch_size reaches its limit?
That would allow us to decide if we need to increase the batch size (and risk bothering the small machine) or send the analytics more often (and start getting messages from machines running in a CI for a single test).

It's not super easy to code since it's located in the segment library, but I think I could do it in a few days :thinking:

Author: I think it's ok for now!

| `infos.max_indexing_memory`                      | Value of `--max-indexing-memory`/`MEILI_MAX_INDEXING_MEMORY` in bytes     | 336042103         | Every Hour |
| `infos.max_indexing_threads`                   | Value of `--max-indexing-threads`/`MEILI_MAX_INDEXING_THREADS` in integer | 4             | Every Hour |
| `infos.ssl_auth_path` | `true` if `--ssl-auth-path`/`MEILI_SSL_AUTH_PATH` is specified, otherwise `false` | false |Every Hour |
| `infos.ssl_cert_path` | `true` if `--ssl-cert-path`/`MEILI_SSL_CERT_PATH` is specified, otherwise `false` | false | Every Hour |
| `infos.ssl_key_path`  | `true` if `--ssl-key-path`/`MEILI_SSL_KEY_PATH` is specified, otherwise `false` | false |Every Hour |
| `infos.ssl_ocsp_path` | `true` if `--ssl-ocsp-path`/`MEILI_SSL_OCSP_PATH` is specified, otherwise `false` | false | Every Hour |
| `infos.ssl_require_auth` | Value of `--ssl-require-auth`/`MEILI_SSL_REQUIRE_AUTH` as a boolean | false | Every Hour |
| `infos.ssl_resumption` | `true` if `--ssl-resumption`/`MEILI_SSL_RESUMPTION` is specified, otherwise `false` | false | Every Hour |

> | `infos.ssl_tickets` | `true` if `--ssl-tickets`/`MEILI_SSL_TICKETS` is specified, otherwise `false` | false | Every Hour |

Author: 📡

| `index_creation`                        | `true` if a document addition or update request triggered index creation in this batch, otherwise `false` | true | `Documents Added`, `Documents Updated` |
| `ranking_rules.words_position`          | Position of the `words` ranking rule if any, otherwise `null` | `1`| `Settings Updated`, `Ranking Rules Updated` |
| `ranking_rules.typo_position`           | Position of the `typo` ranking rule if any, otherwise `null` | `2` | `Settings Updated`, `Ranking Rules Updated` |
| `ranking_rules.proximity_position`      | Position of the `proximity` ranking rule if any, otherwise `null` | `3` | `Settings Updated`, `Ranking Rules Updated` |
| `ranking_rules.attribute_position`      | Position of the `attribute` ranking rule if any, otherwise `null` | `4` | `Settings Updated`, `Ranking Rules Updated` |
| `ranking_rules.sort_position`           | Position of the `sort` ranking rule if any, otherwise `null`. | `5` | `Settings Updated`, `Ranking Rules Updated` |
| `ranking_rules.exactness_position`      | Position of the `exactness` ranking rule if any, otherwise `null`. | `6` | `Settings Updated`, `Ranking Rules Updated` |

> | `ranking_rules.values`                  | An array representing the ranking rules without the custom asc-desc rules. | `['words', 'typo', 'attribute', 'sort', 'exactness']` | `Settings Updated`, `Ranking Rules Updated` |

Reviewer: ```suggestion
| `ranking_rules.values`                  | A string representing the ranking rules without the custom asc-desc rules. | `"words, typo, attribute, sort, exactness"` | `Settings Updated`, `Ranking Rules Updated` |
```

---

## `Documents Deleted`

> > The Documents Deleted event is sent once an hour or when a batch reaches the maximum size of `500kb`. The event's properties are averaged over all requests for the event.

Reviewer: ```suggestion
> The Documents Deleted event is sent once an hour or when a batch reaches the maximum size of `500kb`.
```

I don't think there is any average here right?

## `Settings Updated`

| Property name | Description | Example |
|---------------|-------------|---------|
| user_agent    | Represents the user-agent encountered on this call. | `["Meilisearch Ruby (v2.1)", "Ruby (3.0)"]` |
| searchable_attributes.total | Number of searchable attributes. | `3`|
| searchable_attributes.with_wildcard | `true` if `*` is specified as a searchable attribute, otherwise `false`. | `false` |
| ranking_rules.words_position | Position of the `words` ranking rule if any, otherwise `null` | `1`|
| ranking_rules.typo_position | Position of the `typo` ranking rule if any, otherwise `null` | `2` |
| ranking_rules.proximity_position | Position of the `proximity` ranking rule if any, otherwise `null` | `3` |
| ranking_rules.attribute_position | Position of the `attribute` ranking rule if any, otherwise `null` | `4` |
| ranking_rules.sort_position | Position of the `sort` ranking rule if any, otherwise `null`. | `5` |
| ranking_rules.exactness_position | Position of the `exactness` ranking rule if any, otherwise `null`. | `6` |

> | ranking_rules.values | An array representing the ranking rules without the custom asc-desc rules. | `['words', 'typo', 'attribute', 'sort', 'exactness']` |

Reviewer: ```suggestion
| ranking_rules.values | A string representing the ranking rules without the custom asc-desc rules. | `"words, typo, attribute, sort, exactness"` |
```

| user_agent    | Represents the user-agent encountered on this call. | `["Meilisearch Ruby (v2.1)", "Ruby (3.0)"]` |
| per_index_uid | `true` if an uid is used to fetch an index stat resource, otherwise `false` | `true` |

## `Health Seen`

> > The Health Seen event is sent once an hour or when a batch reaches the maximum size of `500kb`. The event's properties are averaged over all requests on `GET` - `/health`.

Reviewer: ```suggestion
> The Health Seen event is sent once an hour or when a batch reaches the maximum size of `500kb`.
```

And here it's the sum, not the average
